modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/BaseStructuredContentResourceImpl.java

```diff
@@ -592,6 +599,51 @@ public Rating putStructuredContentMyRating(
 		return new Rating();
 	}
 
+	/**
+	 * Invoke this method with the command line:
+	 *
+	 * curl -X 'PUT' 'http://localhost:8080/o/headless-delivery/v1.0/structured-contents/{structuredContentId}/permissions'  -u 'test@liferay.com:test'
+	 */
+	@Override
+	@PUT
+	@Parameters(
+		value = {
+			@Parameter(in = ParameterIn.PATH, name = "structuredContentId")
+		}
+	)
+	@Path("/structured-contents/{structuredContentId}/permissions")
+	@Produces({"application/json", "application/xml"})
+	@Tags(value = {@Tag(name = "StructuredContent")})
+	public void putStructuredContentPermission(
+			com.liferay.portal.vulcan.permissions.Permissions[] permissions,
+			@NotNull @Parameter(hidden = true) @PathParam("structuredContentId")
+				Long structuredContentId)
+		throws Exception {
+
+		PermissionChecker permissionChecker =
+			PermissionThreadLocal.getPermissionChecker();
+
+		String resourceName = getStructuredContentResourceName();
+
+		if (!permissionChecker.hasPermission(
+				0, resourceName, 0, ActionKeys.PERMISSIONS)) {
```

Comment:
I noticed some differences in the generated code, when compared to ours.

In this line, you pass groupId as 0, we pass the entity's groupId. Not sure who is right on this one, yours or ours, or both.

resourceName in our case is `com.liferay.data.engine`, that is, we are checking for the PERMISSIONS action at the portlet level, not the model level (`StructuredContentResourceName`), as you are doing.

Reply:
I tracked the code and groupId was not being used for nothing relevant, I can recheck. Passing the entity is a problem if we want to generate it automatically because we would have to expose a method to get it... or receive it in the path (not all paths do)...

Reply:
About checking portlet or resource... that one is interesting, those are 2 different use cases, right? with 2 different menus... maybe we have to expose both.

Reply:
@nhpatt yeah, all portlets have PERMISSIONS in two levels, we are just checking the top level. groupId is used because permissions can be scoped by Site, by passing 0 we are saying all sites have the same permissions configuration.

Reply:
Mmm... if I remember correctly, portlet permissions need the group, resource permissions need the id of the resource. This solution works for resource permissions but not for portlet.

Reply:
@brunofarache your use case is exactly what we want to cover... so I guess I should leave this endpoint (@Path("/structured-contents/{structuredContentId}/permissions")) for model permissions, no need to use groupId and @Path("/sites/{siteId}/structured-contents/permissions") for portlet permissions with the group... right?